### PR TITLE
5481/fdc3 blotter intents

### PR DESCRIPTION
**Pull request by algreasley**
FDC3 Intent required for Finsemble upgrade
+ exploratory work for OpenFin .. more manual / boilerplate in that context!

### CODE REVIEW

File: src/client/src/App/Trades/CoreFxTrades.tsx [hunk 1]
```diff
@@ -17,6 +27,7 @@ const FxTrades = () => {
     }
     if (window.fdc3) {
       broadcast(context)
```
Comment (algreasley) [inline src/client/src/App/Trades/CoreFxTrades.tsx:29]: the `broadcast` should go away

File: src/client/src/App/Trades/CoreFxTrades.tsx [hunk 2]
```diff
@@ -7,6 +7,16 @@ import { TradesGrid } from "./TradesGrid"
 import { useFxTradeRowHighlight } from "./TradesState"
 import { fxColDef, fxColFields } from "./TradesState/colConfig"
 
+if (window.fdc3) {
+  window.fdc3.addIntentListener("ViewChart", (e) => {
+    console.warn("AG: view chart intent received", e)
```
Comment (algreasley) [inline src/client/src/App/Trades/CoreFxTrades.tsx:12]: succeeds with the "loopback" in the custom provider ..

File: src/workspace/src/dock.ts [hunk 3]
```diff
@@ -48,6 +50,10 @@ export const registerDock = () => {
   return Dock.show()
 }
 
+export async function deregisterdock() {
```
Comment (algreasley) [inline src/workspace/src/dock.ts:53]: incidental mods .. noticed Dock was not quite consistent with Home/Store

File: src/client/vite.config.ts [hunk 4]
```diff
@@ -219,32 +219,43 @@ const injectScriptIntoHtml = (
...
-
-            window.supportsBigInt = typeof BigInt !== 'undefined';
-            window.BigInt = supportsBigInt ? BigInt : function(){};
-          </script>
-          
-          <script async src="https://www.googletagmanager.com/gtag/js?id=${
-            env === "prod" ? "G-Z3PC9MRCH9" : "G-Y28QSEPEC8"
-          }"></script>
-        `,
+    pages: [
```
Comment (algreasley) [inline src/client/vite.config.ts:222]: well, this was a right palaver .. just to get the ["multipage app" config](https://vitejs.dev/guide/build.html#multi-page-app) going was sweat and a few tears

File: src/client/src/OpenFin/apps/Launcher/index.tsx
Comment (algreasley) [inline src/client/src/OpenFin/apps/Launcher/index.tsx:1]: just moved MainRoute into here, like the others ..

File: src/client/public-openfin/rt-fx.json [hunk 5]
```diff
@@ -5,7 +5,8 @@
     "name": "Reactive Trader® FX <ENV_SUFFIX>",
     "icon": "<BASE_URL>/static/media/reactive-trader-icon-256x256.png",
     "url": "<BASE_URL>",
-    "autoShow": false,
+    "providerUrl": "<BASE_URL>/provider.html",
+    "autoShow": true,
```
Comment (algreasley) [inline src/client/public-openfin/rt-fx.json:9]: obvs temp

File: src/client/public-openfin/rt-fx.json [hunk 6]
```diff
@@ -22,12 +23,7 @@
     "version": "<OPENFIN_RUNTIME>",
     "arguments": "--remote-debugging-port=9091"
   },
-  "services": [
-    {
-      "name": "notifications",
-      "manifestUrl": "https://cdn.openfin.co/services/openfin/notifications/1.20.0/app.json"
-    }
-  ],
+  "services": [{ "name": "notifications" }],
```
Comment (algreasley) [inline src/client/public-openfin/rt-fx.json:26]: incidental - just keeps us up to date with Notifications, I don't think we need to peg it to a particular version any more (per OpenFin advice) .. but open to discussion on this

File: src/client/public-openfin/launcher.json [hunk 7]
```diff
@@ -45,12 +40,6 @@
       "alias": "excel-api-addin",
       "version": "2.0.0",
       "forceDownload": true
-    },
-    {
-      "src": "<BASE_URL>/plugin/LimitChecker.zip",
```
Comment (algreasley) [inline src/client/public-openfin/launcher.json:50]: about time this went away now - it has been superseded by the web version

**Review by TobyLi96 — APPROVED** [review]
LGTM

**Pull request merged**